### PR TITLE
feat: git-cliff-release and prepare-pypi-distribution actions, python_bump_and_update_changelog workflow

### PR DESCRIPTION
**Pull request by janbuchar**
See the readme for intended usage.

Any insights about naming are more than welcome :)

### CODE REVIEW

**Review by vdusek — APPROVED** [review]
Just a one question regarding the calling reusable workflows, otherwise, LGTM.

File: release-metadata/README.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Prepare release metadata
+        id: metadata
+        uses: apify/workflows/release-metadata@main
```
Comment (vdusek) [inline release-metadata/README.md:39]: I believe it has to be a path to a workflow file:

```suggestion
        uses: apify/workflows/release-metadata/action.yaml@main
```

File: release-metadata/README.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  test:
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Prepare release metadata
+        id: metadata
+        uses: apify/workflows/release-metadata@main
```
Comment (vdusek) [inline release-metadata/README.md:85]: I believe it has to be a path to a workflow file:

```suggestion
        uses: apify/workflows/release-metadata/action.yaml@main
```

**Review by vdusek — CHANGES_REQUESTED** [review]

File: release-metadata/README.md (hunk 1 above)
Comment (vdusek) [reply release-metadata/README.md:39]: Ok, good, thanks.

**Review by vdusek — APPROVED** [review]
I just don't like the name - maybe `git-cliff-release`? Otherwise LGTM.

**Comment by janbuchar** [comment]
> I just don't like the name - maybe `git-cliff-release`? Otherwise LGTM.

I get that, yeah. `git-cliff-release` is OK for me, too. @B4nan?

**Comment by B4nan** [comment]
I don't mind, on the other hand, such a name sounds like we are exposing git cliff options, wrapping it if you want, which we are not really doing, right? 

Either way, it's an internal thing, the name is not that important, if we ever decide to publish and market this, we can think of something better.

**Comment by janbuchar** [comment]
> I don't mind, on the other hand, such a name sounds like we are exposing git cliff options, wrapping it if you want, which we are not really doing, right?

Eh, if you really want, you can point it to a custom config file... not sure if that counts :smile:

**Comment by B4nan** [comment]
It's not just about that, git cliff is not really about releasing, right? That part is just about the changelog and has nothing to do with releasing really. Maybe we could decouple those two parts, then it would make more sense to have a git cliff action that just alters the changelog file and separate release actions (one per language) that deal with the bumps etc.

But as I said, I am fine with that name.

**Comment by janbuchar** [comment]
> It's not just about that, git cliff is not really about releasing, right? That part is just about the changelog and has nothing to do with releasing really. Maybe we could decouple those two parts, then it would make more sense to have a git cliff action that just alters the changelog file and separate release actions (one per language) that deal with the bumps etc.
> 
> But as I said, I am fine with that name. 

The action also gives you the recommended version, but yeah, that point is valid. The biggest point in favor of git-cliff-release is IMO that we'll call it "that git cliff release action" anyway 😃

**Pull request merged**